### ISSUE TITLE
Pyright should not allow treating PEP 695 type aliases as PEP 484 ones

### ISSUE THREAD

**Issue by LeeeeT** [microsoft/pyright#5662]
**Describe the bug**
[PEP 484 type aliases](https://peps.python.org/pep-0484/#type-aliases) are simple variables with the exact same object as the original type.
However, the [new syntax introduced in PEP 695](https://peps.python.org/pep-0695/#generic-type-alias) allows creating type aliases like this: `type a = int`. Which creates a new object of type `typing.TypeAliasType`. And this new object cannot be used as the original type at runtime (e.g. accessing its attributes or inheriting from it).

**Code or Screenshots**
```python
type MyInt = int

MyInt.bit_count(1)

class DerivedInt(MyInt): pass
```

This code produces an error at runtime but is completely fine for pyright. I guess treating PEP 695 type aliases as PEP 484 ones would be a common mistake (especially trying to inherit from them) and I expect pyright to give me a static type checking error when doing that.

pyright 1.1.320

**Comment by erictraut** [comment]
I think this is a reasonable enhancement request.

**Comment by erictraut** [comment]
This will be included in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.321, which I just published. It will also be included in a future release of pylance.
